### DISCUSSION TITLE
Is it possible to use the OffscreenCanvas Web API?

### DISCUSSION THREAD

**Discussion by marcoripa96** [jeffreylanters/react-unity-webgl#535] [Question and Answer]
As the title question suggest, I came across this web API https://developer.mozilla.org/en-US/docs/Web/API/OffscreenCanvas, is it possible to have the webgl computation in a web worker?

It looks like the `transferControlToOffscreen` function must be called before getting the rendering context of the canvas.

**Comment by ilmarioranen** [comment]
I think the using OffscreenCanvas to offload Unity WebGL CPU burdens to a different worker is critically important feature - especially during the initial heavy loading phase. Without this feature Unity WebGL can and will make other JS logic and UI unresponsive, which pretty much pisses of every user and ruins the experience.

It's not impossible to pull this off, but apparently not easy either. Readers of this issue probably already found this example of a case where it was accomplished with Unity WebGL in 2019:
[https://discussions.unity.com/t/unity-in-worker-using-transfercontroltooffscreen/732599/6](https://discussions.unity.com/t/unity-in-worker-using-transfercontroltooffscreen/732599/6)
